source/features/collapse-markdown-sections.ts

```diff
@@ -0,0 +1,48 @@
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+function toggleSection(sectionHeading: HTMLElement): void {
+	const sectionState = sectionHeading.classList.contains('rgh-markdown-section-collapsed');
+	sectionHeading.classList.toggle('rgh-markdown-section-collapsed');
+
+	let element = sectionHeading.nextElementSibling as HTMLElement;
+	while (element) {
+		if (element.matches('h1, h2, h3, h4, h5, h6')) {
+			if (sectionHeading.tagName.localeCompare(element.tagName) >= 0) {
+				return;
+			}
+
+			element.classList.toggle('rgh-markdown-section-collapsed', !sectionState);
+		} else {
+			element.hidden = !sectionState;
+		}
+
+		element = element.nextElementSibling as HTMLElement;
+	}
+}
+
+function toggleAllTopSections(): void {
+	for (const topHeading of select.all('.markdown-body > h1')) {
+		toggleSection(topHeading);
+	}
+}
+
+function toggleSingleSection({delegateTarget: sectionHeading}: delegate.Event<MouseEvent, HTMLElement>): void {
+	toggleSection(sectionHeading);
+}
+
+function init(): void {
+	delegate(document, '.markdown-body > h1', 'click', toggleAllTopSections);
+	delegate(document, '.markdown-body > :is(h2, h3, h4, h5, h6)', 'click', toggleSingleSection);
+}
+
+void features.add(__filebasename, {
+	include: [
+		pageDetect.isRepoRoot,
```

Comment:
Not just the root, but any folder that has a readme. Testable in any monorepo like Babel